tests/integration-tests/tests/budgets/test_budgets.py

```diff
@@ -33,6 +36,44 @@ def test_update_budget_properties(pcluster_config_reader, clusters_factory, test
     # Add one budget of each BudgetCategory, update them, and check that the fields match the expected values
     _test_update_cluster_and_check_fields(cluster, budget_client, account_id, region, pcluster_config_reader)
 
+    # Enable fleet stop action, trigger sns topic notification, and assert that the fleet has stopped
+    _test_stop_compute_fleet(cluster, pcluster_config_reader)
+
+
+def _test_stop_compute_fleet(cluster, pcluster_config_reader):
+    # Create the initial cluster with a budget triggered compute-fleet stop action
+    stop_fleet_config_file = pcluster_config_reader(config_file="budget_with_fleet_stop.yaml")
+    cluster.update(str(stop_fleet_config_file), force_update="true")
+
+    cluster_info = cluster.describe_cluster()
+
+    assert_that(cluster_info.get("computeFleetStatus")).is_in("RUNNING", "ENABLED")
+
+    sns_client = boto3.client("sns")
+
+    # Look for the right topic using boto3
+    topic_name = f"pcluster-BudgetTopic-{cluster.name}-{cluster.region}"
+    topic_list = sns_client.list_topics().get("Topics")
+    budget_topic_arn = None
+    for topic in topic_list:
```

Comment:
We talked about budget arn is i the format of : 
`arn:{partition}:sns:{cluster.region}:{account_id}:{topic_name}`
where:
```
partition = next(("aws-" + partition for partition in ["us-gov", "cn"] if cluster.region.startswith(partition)), "aws")
account_id = boto3.client("sts", region_name=cluster.region).get_caller_identity().get("Account")
```

Then we don't need to retrieve the budget arn by listing all budget and get the arn from name

Reply:
Good idea. Updated!